rp2040-hal/src/rtc.rs

```diff
@@ -1,3 +1,620 @@
+        self.day_of_week = Some(day_of_week);
+        self
+    }
+    /// Set a filter on the given hour
+    pub fn hour(mut self, hour: u8) -> Self {
+        self.hour = Some(hour);
+        self
+    }
+    /// Set a filter on the given minute
+    pub fn minute(mut self, minute: u8) -> Self {
+        self.minute = Some(minute);
+        self
+    }
+    /// Set a filter on the given second
+    pub fn second(mut self, second: u8) -> Self {
+        self.second = Some(second);
+        self
+    }
+}
+
+impl RtcClock {
+    /// Init the RTC clock. This should be called exactly once at the start of the application.
+    ///
+    /// [`init_clocks_and_plls`] will call this internally so you often won't have to call this.
+    ///
+    /// [`init_clocks_and_plls`]: fn.init_clocks_and_plls.html
+    pub fn init(&mut self, resets: &mut RESETS) {
+        // reset RTC
+        resets.reset.modify(|_, w| w.rtc().set_bit());
+        resets.reset.modify(|_, w| w.rtc().clear_bit());
+        while resets.reset_done.read().rtc().bit_is_clear() {
+            core::hint::spin_loop();
+        }
+        // Set the RTC divider
+        let freq = self.freq().integer() - 1;
+        unsafe { &*RTC::ptr() }
+            .clkdiv_m1
+            .write(|w| unsafe { w.bits(freq) });
+    }
+    /// Checks to see if this RtcClock is running
+    pub fn is_running(&mut self) -> bool {
+        // Safety: We only read from the given address
+        let rtc = unsafe { &*RTC::ptr() };
+        let ctrl = rtc.ctrl.read().bits();
+        (ctrl & consts::RTC_CTRL_RTC_ACTIVE_BITS) > 0
+    }
+
+    /// Set the RTC clock to the given DateTime. After this is set, the RtcClock will automatically keep the date synchronized.
+    ///
+    /// To retrieve the current DateTime, see [`now`]
+    ///
+    /// This DateTime persists through sleep mode and dormant mode, but not through chip resets.
+    ///
+    /// Note that the datetime calculation may be incorrect, especially when dealing with leapyears.
+    /// See [Chapter 4 Section 8](https://datasheets.raspberrypi.org/rp2040/rp2040_datasheet.pdf) for more details
+    ///
+    /// [`now`]: #method.now
+    pub fn set_datetime(&mut self, t: &DateTime) -> Result<(), RtcError> {
+        t.validate().map_err(RtcError::InvalidDateTime)?;
+
+        // NOTE: No functions should be called in this block as we take an unsafe ownership of `RTC::ptr()`
+        {
+            // Safety: We should have unique access to this RtcClock because we have `&mut self`
+            // and this file is the only file that messes with these addresses.
+            let rtc = unsafe { &*RTC::ptr() };
+
+            rtc.ctrl.modify(|_, w| w.rtc_enable().clear_bit());
+            // TODO: Add a timeout to this?
+            while rtc.ctrl.read().rtc_active().bit_is_set() {
+                core::hint::spin_loop();
+            }
+
+            let setup_0 = ((t.year as u32) << consts::RTC_SETUP_0_YEAR_LSB)
```

Comment:
I think there are suitable PAC accessor methods available, so you shouldn't need the consts. See https://docs.rs/rp2040-pac/0.2.0/rp2040_pac/rtc/setup_0/index.html.